### PR TITLE
add cassido problem for week of: Nov 18th 2024

### PR DESCRIPTION
**Pull request by Barbacoa08**
## Summary by CodeRabbit

- **New Features**
	- Introduced a new function, `MaximizeStocks`, to calculate the maximum profit from stock prices with an optimized implementation.
	- Added scripts for building and testing the "maximizestocks" component, enhancing the development workflow.
	- Created a new markdown file detailing requirements and usage examples for the `maxTheStock` function.

- **Tests**
	- Established a comprehensive test suite for the `MaximizeStocks` function, covering various scenarios including edge cases.

- **Documentation**
	- Added a markdown file outlining requirements for the stock profit calculation function.

### CODE REVIEW

File: src/maximize-stocks/solution/maximize-stocks.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+		for (let j = i; j < prices.length; j++) {
+			const possibleGap = prices[j] - uut;
+			if (possibleGap > gap) {
+				gap = possibleGap;
+			}
+		}
+	}
+
+	return gap;
+};
```
[the remark this answers is not in the document]
Comment (Barbacoa08) [reply src/maximize-stocks/solution/maximize-stocks.ts]: amazing, thanks!
